### PR TITLE
Issue #6

### PR DESCRIPTION
**Pull request by HaydenHoran**
Hi, I have created an addition function example (in javascript) that can be executed in a web browser via an html document (I have written this code as well). Both files have been a copious amount of comments.

I also have a drawing example using processing.js if you are interested.
Thanks!

### CODE REVIEW

**Comment by Light---** [comment]
@HaydenHoran Hey; Thanks for putting up the awesome PR! :+1:

Would you mind squashing down some of the commits? You can find the steps here : https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

File: js/BasicJSExamples/Addition.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+/* Another useful note: Notice how my function and variables were given a name with meaning. If I had given them arbitrary 
+names, it would be difficult to know what information they contained. For example, if I had named my function "elephant",
+I would not know for what the function is actually used.
+There are a few ways we can execute this code: using node.js, programming.js, or using html are probably the three most common.
+We will do the latter. Please see ExecuteAdditionFunction.html (in this directory) for details.
+*/
+
+
```
Comment (Light---) [inline js/BasicJSExamples/Addition.js:80]: Mind deleting this whitespace?

File: js/BasicJSExamples/Addition.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+acts on the argument (an integer) that is placed before the period.
+*/
+var myString= 'The addition of the two numbers is'
+//We make another string
+
+var Space= ' '
+//We make a space. Note, this is also a string.
+
+var Output= myString+Space+convertSumtoString
+/* This is interesting. We notice that we can "add" strings together! Just like we can add integers.
```
Comment (Light---) [inline js/BasicJSExamples/Addition.js:61]: We could also introduce the word `concatenation` here maybe? What are your thoughts 💭

File: js/BasicJSExamples/Addition.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+However, we cannot subtract, multiply, or divide strings, for obvious reasons. 
+*/
+
+return (Output);
+/*The return value is of pivotal importance. After the function (in this case Addition) accepts its inputs, 
+and operates on those inputs (adds them together then converts them into a string), the return keyword is where the 
+final value is stored. This value can be passed to other functions, where further manipulations can be made, as we will see.
```
Comment (Light---) [inline js/BasicJSExamples/Addition.js:68]: Maybe we could reword this to

> the return keyword is used to determine the resulting value of the function.

File: js/BasicJSExamples/Addition.js (hunk 1 above)
Comment (HaydenHoran) [inline js/BasicJSExamples/Addition.js:80]: I will get on this when I can!
I am desperately trying to find housing for the summer, so it may take a
little while.

File: js/BasicJSExamples/Addition.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+*/
+
+}
+
+/* Another useful note: Notice how my function and variables were given a name with meaning. If I had given them arbitrary
```
Comment (Light---) [inline js/BasicJSExamples/Addition.js:73]: :+1: Great point here on naming conventions

File: js/BasicJSExamples/Addition.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+tell our machine to make it a function by writing "= function".
+*/
+
+var Sum= a+ b;
+//We define another variable "Sum" that stores the value of 'a' + 'b'.
+
+var convertSumtoString= Sum.toString();
+/* For reasons that will become apparent, we want to convert our variable "Sum" into a string.
+//The function .toString is an example of a predefined function in the JavaScript language, which 
+acts on the argument (an integer) that is placed before the period.
```
Comment (Light---) [inline js/BasicJSExamples/Addition.js:52]: Minor nitpick here - JavaScript doesn't have integers! Instead everything is actually a [`number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number)

Here's a quick example -

``` javascript
> typeof 13.37
"number"
```

``` javascript
> typeof 13
"number"
```

File: js/BasicJSExamples/Addition.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+All programming languages contain functions. They are the atomistic elements of programming.
+The first function that you will probably come accross in all languages is the "adding function".
+That is exactly the function that will be presented here, to demonstrate the (very) basic 
+uses of JavaScript.
+*/
+
+/*Commenting: Whenever you see a "//" or "/*", note that it does not contribute to the 
+functionality of the code. However, commenting is a very important part of programming,
+especially when the code becomes very complex. It gives the opportunity to the programmer
+to remind himself and tell others(like you!) what the code does, what variables define, etc.
```
Comment (Light---) [inline js/BasicJSExamples/Addition.js:11]: We could go further and say about how it defined a contract to developers too - referred to as an [API](https://en.wikipedia.org/wiki/Application_programming_interface)

> It gives the opportunity to future developers to understand what arguments the function can receive, and what values can be expected to be returned. This can be referred to as an Application Programming Interface, or API for short.

File: js/BasicJSExamples/Addition.js (hunk 1 above)
Comment (Light---) [inline js/BasicJSExamples/Addition.js:80]: @HaydenHoran That's no problem; Take your time! :+1:

**Comment by Light---** [comment]
@HaydenHoran Awesome work; Let me know if you need any clarifications :+1:
